plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -441,6 +481,56 @@ public void setNull(PreparedStatement statement, int index)
         };
     }
 
+    public static ColumnMapping longTimestampColumnMapping(TimestampType type)
+    {
+        if (type.getPrecision() <= MAX_SHORT_PRECISION) {
+            return timestampColumnMapping(type);
```

Comment:
we should add a validation into standard read function too. This can be a followup but please add an issue for that.

Reply:
Added!